src/bin/scheduler/setup/directories.rs

```diff
@@ -56,18 +56,87 @@ pub fn setup(
     create_dir_all(&global_config.managed_directory)?;
 
     setup_results_directories(global_config, &plans, &ownership_setter)?;
+    #[cfg(windows)]
+    create_dir_all(&global_config.robocorp_home_base)?;
+
     let (surviving_plans, managed_dir_failures) = setup_managed_directories(plans);
+    #[cfg(windows)]
+    let (surviving_plans, robocorp_home_failures) =
+        setup_robocorp_home_directories(global_config, surviving_plans);
     let (mut surviving_plans, working_dir_failures) =
         setup_working_directories(global_config, surviving_plans, &ownership_setter);
 
     sort_plans_by_grouping(&mut surviving_plans);
-    Ok((
-        surviving_plans,
-        managed_dir_failures
-            .into_iter()
-            .chain(working_dir_failures)
-            .collect(),
-    ))
+    #[cfg(windows)]
+    let failures = managed_dir_failures
+        .into_iter()
+        .chain(working_dir_failures)
+        .chain(robocorp_home_failures)
+        .collect();
+    #[cfg(unix)]
+    let failures = managed_dir_failures
+        .into_iter()
+        .chain(working_dir_failures)
+        .collect();
+    Ok((surviving_plans, failures))
+}
+
+#[cfg(windows)]
+fn setup_robocorp_home_directories(
+    global_config: &GlobalConfig,
+    plans: Vec<Plan>,
+) -> (Vec<Plan>, Vec<SetupFailure>) {
+    use super::windows_permissions::grant_full_access;
+    use log::info;
+    use robotmk::session::Session;
+
+    let mut surviving_plans = vec![];
+    let mut failures = vec![];
+
+    for (session, plans_in_session) in plans_by_sessions(plans) {
+        let robocorp_home = session.robocorp_home(&global_config.robocorp_home_base);
```

Comment:
We should take this value from `RCCEnvironment`, not re-compute it. See also `setup_environment_building_directories`.

Reply:
Hmm, this would require us to call `plans_in_session[0].environment` no? Are we really sure that the plan `RCCEnvironment` is the same environment used to do the initial configuration? I think it is better keep these two constructions independent.